src/main/java/org/radarbase/appserver/service/FcmNotificationService.java

```diff
@@ -237,6 +270,25 @@ public FcmNotificationDto updateNotification(
         return notificationConverter.entityToDto(notificationSaved);
     }
 
+    @Transactional
+    public FcmNotifications scheduleAllUserNotifications(String subjectId, String projectId) {
+
+        User user = subjectAndProjectExistElseThrow(subjectId, projectId);
+        List<Notification> notifications = notificationRepository.findByUserId(user.getId());
+        this.schedulerService.scheduleMultiple(notifications);
```

Comment:
we should check if the notification is already scheduled or not. Otherwise, a single notification can be delivered multiple times.

Possibly, it should not work if one already exists for the id (https://github.com/RADAR-base/RADAR-Appserver/blob/cfc14705850ebf5bb40b43e640279b988e168439/src/main/java/org/radarbase/appserver/service/scheduler/MessageSchedulerService.java#L96) but please confirm in Quartz scheduler docs.

Reply:
Hi Yatharth,

I can see we have to do it while scheduling single job but I am not sure if we need to do it on batch scheduling. The reason is that  according to https://www.quartz-scheduler.org/api/2.1.7/org/quartz/Scheduler.html#scheduleJobs(java.util.Map,%20boolean), the jobs which are already scheduled got replaced using scheduleJobs function.

Reply:
Ok yes, if it fails on the same key then we don't need to check. It should then fail similarly for single jobs too?

Reply:
Although it might be worth catching the [ObjectAlreadyExistsException](https://www.quartz-scheduler.org/api/2.1.7/org/quartz/ObjectAlreadyExistsException.html) and returning the appropriate response (to tell the client that the notification already exists).